src/bika/lims/browser/worksheet/tools.py

```diff
@@ -93,3 +95,15 @@ def getWorksheetLayouts():
         map(lambda entry: layouts.append(entry), layout_utility.getLayouts())
 
     return DisplayList(tuple(layouts))
+
+
+def getServiceUidsByMethod(method):
+    query = {
+        "portal_type": "AnalysisService",
+        "is_active": True,
+        "method_available_uid": method,
+    }
+    setup_catalog = api.get_tool(SETUP_CATALOG)
+    uids = map(lambda s: s.UID, setup_catalog(query))
+
+    return uids
```

Comment:
Please move this function to `add_analyses.py`. Also note you assume the value for `method` is a always UID. Worth to check and resolve the value for `method` first.

Reply:
done